### PR TITLE
refactor: type safety and formatting

### PR DESCRIPTION
**Pull request by jwagenleitner**

### CODE REVIEW

File: src/main/org/codehaus/groovy/transform/trait/Traits.java [hunk 1]
```diff
@@ -355,19 +355,19 @@ static String getSuperTraitMethodName(ClassNode trait, String method) {
...
-    public static @interface Implemented {}
+    public @interface Implemented {}
 
     /**
      * Internal annotation used to indicate that a method is a bridge method to a trait
      * default implementation.
      */
     @Retention(RetentionPolicy.RUNTIME)
     @Target(ElementType.METHOD)
-     public static @interface TraitBridge {
```
Comment (graemerocher) [inline src/main/org/codehaus/groovy/transform/trait/Traits.java]: This is a breaking change, some libraries like GORM needs to inspect this information in order to know whether a property or fields i provided by a trait. See https://github.com/grails/grails-data-mapping/blob/master/grails-datastore-core/src/main/groovy/org/grails/datastore/mapping/reflect/FieldEntityAccess.java#L227

File: src/main/org/codehaus/groovy/transform/trait/Traits.java [hunk 2]
```diff
@@ -355,19 +355,19 @@ static String getSuperTraitMethodName(ClassNode trait, String method) {
      */
     @Retention(RetentionPolicy.RUNTIME)
     @Target(ElementType.METHOD)
-    public static @interface Implemented {}
+    public @interface Implemented {}
```
Comment (graemerocher) [inline src/main/org/codehaus/groovy/transform/trait/Traits.java]: Also breaking. See https://github.com/grails/grails-data-mapping/blob/master/grails-datastore-core/src/main/groovy/org/grails/datastore/mapping/reflect/FieldEntityAccess.java#L504

File: src/main/org/codehaus/groovy/transform/trait/Traits.java (hunk 1 above)
Comment (jwagenleitner) [inline src/main/org/codehaus/groovy/transform/trait/Traits.java]: The `static` modifier is redundant on inner interfaces. The bytecode will remain unchanged with or without the modifier so the code referenced should continue to work.

File: src/main/org/codehaus/groovy/transform/trait/Traits.java (hunk 2 above)
Comment (graemerocher) [inline src/main/org/codehaus/groovy/transform/trait/Traits.java]: Ah of course.. silly me

**Comment by paulk-asert** [comment]
+1

**Pull request closed without merging**